examples/gno.land/r/gov/dao/dao.gno

```diff
@@ -61,29 +66,48 @@ func VoteOnProposal(idx int, option string) {
 	AssertIsMember(caller)
 
 	prop := getProposal(idx)
-	if prop.finished {
+
+	if prop.executor.IsExpired() {
+		panic("prop is expired, cannot vote.")
+	}
+
+	if prop.voter.Finished(members) {
 		panic("prop is not active anymore, cannot vote.")
 	}
-	// XXX: implement the real voting (would be cool to have a generic p/)
-	prop.accepted = option == "YES"
-	prop.finished = true
+
+	prop.voter.Vote(members, option)
 }
 
 func ExecuteProposal(idx int) {
 	assertProposalExists(idx)
 	prop := getProposal(idx)
-	if !prop.finished {
+
+	if prop.executor.IsExpired() {
+		panic("prop is expired, cannot execute.")
+	}
+
+	if !prop.voter.Finished(members) {
 		panic("prop is still active, cannot execute.")
 	}
-	if !prop.accepted {
+	if !prop.voter.Accepted(members) {
 		panic("prop is not accepted, cannot execute.")
 	}
+
 	prop.executor.Execute()
 }
 
 func IsMember(addr std.Address) bool {
-	// XXX: implement
-	return true // in the meantime, everyone is a DAO member
+	if len(members) == 0 { // special case for initial execution
```

Comment:
Do we still want to keep this?

Reply:
Yep, only needed for the first proposal, usually the one adding members to the DAO.